File: server/src/main/java/server/models/Event.java
package server.models;

import jakarta.persistence.Entity;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.JoinTable;
import jakarta.persistence.ManyToMany;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;
import java.util.Objects;

@Entity
public class Event extends CalendarEntry {
    @ManyToMany
    @JoinTable(
            name = "event_user",
            joinColumns = @JoinColumn(name = "event_id"),
            inverseJoinColumns = @JoinColumn(name = "user_id")
    )
    private List<User> attendees = new ArrayList<>();

    public Event(Project project, String name, String description, Date startDate, Date endDate, List<User> attendees) {
        super(project, name, description, startDate, endDate);
        this.attendees = attendees;
    }

    public List<User> getAttendees() {
        return attendees;
    }

    public void setAttendees(List<User> attendees) {
        this.attendees = attendees;
    }

    @Override
    public String toString() {
        return "Event{" +
                super.toString() +
                ", attendees=" + attendees +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        if (!super.equals(o)) return false;
        Event event = (Event) o;
        return Objects.equals(attendees, event.attendees);
    }

    @Override
    public int hashCode() {
        return Objects.hash(super.hashCode(), attendees);
    }
}

File: client/src/types/Event.ts
import { ProjectResponse } from "../features/project/projectService";

export type Event = {
  id: number;
  name: string;
  description: string;
  startDate: Date;
  endDate: Date;
  createdAt: string;
};

export type Task = {
  id: number;
  name: string;
  description: string;
  startDate: Date;
  endDate: Date;
  createdAt: string;
  status: string;
  project: ProjectResponse;
  subTasks: SubTask[];
};

export type SubTask = {
  id: number;
  name: string;
  desc: string;
};
